pac4j-saml/src/main/java/org/pac4j/saml/metadata/SAML2IdentityProviderMetadataResolver.java

```diff
@@ -71,27 +75,31 @@ public final MetadataResolver resolve() {
                     path = "/" + path;
                 }
                 resource = ResourceHelper.of(new ClassPathResource(path));
-            } else if (this.idpMetadataPath.startsWith("file:")) {
-                resource = ResourceHelper.of(new FileSystemResource(this.idpMetadataPath));
-            } else if (this.idpMetadataPath.startsWith("http")) {
+            }  else if (this.idpMetadataPath.startsWith("https")) {
```

Comment:
Metadata needs to come from an https source, so that we ca have a bit of extra security via SSL. A URL metadata typically comes from an SP endpoint itself, and the SP must be protected via https. If your app isn't secure, it's really not worth SSO'ing into :)